lib/index.js

```diff
@@ -189,12 +194,11 @@ const isGitHubShorthand = (arg) => {
 // attempt to correct an scp style url so that it will parse with `new URL()`
 const correctUrl = (giturl) => {
   const firstAt = giturl.indexOf('@')
-  const lastHash = giturl.lastIndexOf('#')
-  let firstColon = giturl.indexOf(':')
-  let lastColon = giturl.lastIndexOf(':', lastHash > -1 ? lastHash : Infinity)
+  // ignore colons that come after the hash since that could include colons such as:
+  // git@github.com:user/package-2#semver:^1.0.0
+  const lastColonBeforeHash = lastIndexOfBefore(giturl, ':', '#')
 
-  let corrected
```

Comment:
I removed the `corrected` variable declaration because we were assigning to it in the first `if` block but then not guarding against it in the second `if` block despite using it there. So it's effectively the same as mutating `giturl`.